localstack/aws/app.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, service_manager: ServiceManager = None) -> None:
             [
                 handlers.push_request_context,
                 metric_collector.create_metric_handler_item,
+                handlers.preprocess_request,
```

Comment:
As soon as the legacy S3 hander is removed and we added a `preprocess_request` handler for the API GW CORS handling, we could move the service name detection even below the CORS handler, right? :)

Reply:
Exactly! While trying it and testing it, having the service name detection below the CORS handler was working fine (except API GW). That is also why we set the service in the context already. I guess it is not useful yet, but it is already there for the next iteration.

Reply:
And with this solution, even the internal endpoints would be served with default CORS routes as the "bucket" wouldn't exist and it wouldn't be match as an `s3` request. This seems to solve our different issues with the service name parser.